api-service/src/controllers/DatasetUpdate/DatasetUpdate.ts

```diff
@@ -0,0 +1,297 @@
+
+            if (!_.isEmpty(deleteTransformation)) {
+                await DatasetTransformationsDraft.destroy({ where: { id: deleteTransformation } });
+            }
+        }
+    }
+}
+
+const checkDatasetExists = async (dataset_id: string): Promise<Record<string, any>> => {
+    const datasetExists: Record<string, any> | null = await getExistingDataset(dataset_id)
+    if (datasetExists) {
+        return { isDatasetExists: true, datasetStatus: datasetExists.status }
+    } else {
+        return { isDatasetExists: false, datasetStatus: "" }
+    }
+}
+
+const getUpdatedConfigs = async (payload: Record<string, any>): Promise<Record<string, any>> => {
+    const { validation_config, extraction_config, dedup_config, tags, denorm_config, dataset_id } = payload
+    const existingDataset: any = await getExistingDataset(dataset_id)
+    const updatedConfigs = {
+        validation_config: validation_config ? setValidationConfigs(validation_config) : null,
+        extraction_config: extraction_config ? setExtractionConfigs(extraction_config) : null,
+        dedup_config: dedup_config ? setDedupConfigs(dedup_config) : null,
+        tags: tags ? getUpdatedTags(tags, _.get(existingDataset, "tags")) : null,
+        denorm_config: denorm_config ? setDenormConfigs(denorm_config, _.get(existingDataset, ["denorm_config"])) : null
+    }
+    return _.pickBy({ ...payload, ...updatedConfigs }, _.identity)
+}
+
+const getDuplicateDenormKey = (denormConfig: Record<string, any>): Array<string> => {
+    if (denormConfig && _.isArray(_.get(denormConfig, 'denorm_fields'))) {
+        const denormFields = _.get(denormConfig, "denorm_fields")
+        const denormOutKeys = _.compact(_.map(denormFields, field => _.get(field, "action") == "add" && _.get(field, "values.denorm_out_field")))
+        const duplicateDenormKeys: Array<string> = _.filter(denormOutKeys, (item: string, index: number) => _.indexOf(denormOutKeys, item) !== index);
+        return duplicateDenormKeys;
+    }
+    return []
+}
+
+const setValidationConfigs = (configs: Record<string, any>): Record<string, any> => {
+    const validationStatus = _.get(configs, "validate")
+    if (!validationStatus) {
+        _.set(configs, "mode", defaultDatasetConfig.validation_config.mode)
+    }
+    return configs;
+}
+
+const setExtractionConfigs = (configs: Record<string, any>): Record<string, any> => {
+    const isBatchEvent = _.get(configs, "is_batch_event")
+    if (!isBatchEvent) {
+        return _.merge(configs, defaultDatasetConfig.extraction_config)
+    }
+    return configs
+}
+
+const setDedupConfigs = (configs: Record<string, any>): Record<string, any> => {
+    const dropDuplicates = _.get(configs, "drop_duplicates")
+    if (!dropDuplicates) {
+        return { ...defaultDatasetConfig.dedup_config, drop_duplicates: dropDuplicates }
+    }
+    return configs
+}
+
+const getUpdatedTags = (payload: Record<string, any>, datasetTags: Array<string>): Record<string, any> => {
```

Comment:
Error scenarios
` action === "add" || "remove"
ptags = lower(pTags)
 pTags = uniq(pTags)

1. Action == "add" && includes()
2. Action ==. "remove" && !included()

1. action === "add" {
   tags = _.lower([..existing , ...provided])
  // update the tags in table
}
1. action === "remove" {
   tags = _.lower(filter(tags, pTags))
  // update the tags in table
}`